examples/pendulum/pendulum_nn.py

```diff
@@ -71,53 +71,44 @@ def dynamics_batch_nn(x, u):
     xnext = dynamics_net(xu)
     return xnext.detach().numpy()    
 
-def jacobian_x_nn(x, u):
+def jacobian_xu_nn(x, u):
     xu = torch.Tensor(np.concatenate((x,u))).unsqueeze(0)
     xu.requires_grad = True
     xnext = dynamics_net(xu)
     dJdxu0 = torch.autograd.grad(xnext[0,0], xu, retain_graph=True)[0].numpy()
     dJdxu1 = torch.autograd.grad(xnext[0,1], xu)[0].numpy()
     dJdxu = np.vstack((dJdxu0, dJdxu1))
-    return dJdxu[0:2, 0:2]
-
-def jacobian_u_nn(x, u):
-    xu = torch.Tensor(np.concatenate((x,u))).unsqueeze(0)
-    xu.requires_grad = True
-    xnext = dynamics_net(xu)
-    dJdxu0 = torch.autograd.grad(xnext[0,0], xu, retain_graph=True)[0].numpy()
-    dJdxu1 = torch.autograd.grad(xnext[0,1], xu)[0].numpy()
-    dJdxu = np.vstack((dJdxu0, dJdxu1))
-    return dJdxu[0:2, 2, None]
+    return dJdxu[0:2]
 
 """5. Test SQP."""
 timesteps = 200
-Q = np.diag([5, 5])
-R = 0.1 * np.diag([1])
+Q = np.diag([1, 1])
+Qd = np.diag([20., 20.])
+R = np.diag([1])
 x0 = np.array([0, 0])
 xd = np.array([np.pi, 0])
 xdt = np.tile(xd, (timesteps+1,1))
 xbound = [
-    -np.array([4.0 * np.pi, 20.0]),
-     np.array([4.0 * np.pi, 20.0])
+    -np.array([1e4, 1e4]),
+     np.array([1e4, 1e4])
 ]
 ubound = np.array([
-    -np.array([20.0]),
-     np.array([20.0])
+    -np.array([1e4]),
+     np.array([1e4])
 ])
 
 # 3. Set up initial guess.
 u_trj = np.tile(np.array([0.1]), (timesteps,1))
-x_initial_var = np.array([10.0, 40.0])
-u_initial_var = np.array([40.0])
+x_initial_var = np.array([1.0, 1.0])
+u_initial_var = np.array([1.0])
 num_samples = 10000
 
 # 4. Solve.
 try:
-    sqp_exact = SQP_Exact_Explicit(
+    sqp_exact = DiLQR_Exact(
         dynamics_nn,
-        jacobian_x_nn,
-        jacobian_u_nn,
-        Q, R, x0, xdt, u_trj,
+        jacobian_xu_nn,
+        Q, Qd, R, x0, xdt, u_trj,
```

Comment:
`xdt` always makes me think of the time derivative of `x`. `x_trj_d` (desired x trajectory) could be a better name?

Reply:
This has been fixed in the class, but I will clean up naming convention for the scripts as well.

Reply:
And here `u_trj` is the initial guess for the `u` trajectory? Maybe we can make it more explicit by calling it `u_trj_initial` or `u_trj_0`?